model/licensing.py

```diff
@@ -567,6 +645,57 @@ def needs_update(self):
         age = now - self.last_checked
         return age > maximum_stale_time
 
+    def update_availability_from_licenses(
+        self,
+        analytics: "Analytics" = None,
+        as_of: datetime.datetime = None,
+    ):
+        """
+        Update the LicensePool with new availability information, based on the
+        licenses and holds that are associated with it.
+
+        Log the implied changes with the analytics provider.
+        """
+        _db = Session.object_session(self)
+
+        licenses_owned = sum([l.total_remaining_loans for l in self.licenses])
+        licenses_available = sum([l.currently_available_loans for l in self.licenses])
+
+        holds = self.get_active_holds()
+
+        patrons_in_hold_queue = len(holds)
+        if len(holds) > licenses_available:
+            licenses_reserved = licenses_available
+            licenses_available = 0
+        else:
+            licenses_reserved = len(holds)
+            licenses_available -= licenses_reserved
+
+        return self.update_availability(
+            licenses_owned,
+            licenses_available,
+            licenses_reserved,
+            patrons_in_hold_queue,
+            analytics=analytics,
+            as_of=as_of,
+        )
+
+    def get_active_holds(self):
+        _db = Session.object_session(self)
+        return (
+            _db.query(Hold)
+            .filter(Hold.license_pool_id == self.id)
+            .filter(
+                or_(
+                    Hold.end == None,
+                    Hold.end > utc_now(),
+                    Hold.position > 0,
```

Comment:
I'm not 💯 on holds in ODL yet. Holds do work in the testing I've done, but I'm not sure if all the edge cases are covered. So we may have to do some further refactoring for holds.

My thinking with this code was that we already had accounted for holds with position > 0 as they were counted in the `licenses_reserved` on the `LicensePool`. But giving this another look over to try to answer your question, I'm not sure that is correct now.

Reply:
I'm thinking our license accounting is better with these changes then without, and it will unblock some things, so if it makes sense to you I'd like to proceed with this PR, and take a bit of time to dive a bit deeper into how the hold queue is working and make sure its consistent, as a follow up to this.

Reply:
💯